include/boost/geometry/algorithms/detail/envelope/multipoint.hpp

```diff
@@ -0,0 +1,362 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2015, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Menelaos Karavelas, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_ENVELOPE_MULTIPOINT_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_ENVELOPE_MULTIPOINT_HPP
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include <boost/algorithm/minmax_element.hpp>
+#include <boost/range.hpp>
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/coordinate_system.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+#include <boost/geometry/core/tags.hpp>
+
+#include <boost/geometry/util/convert_on_spheroid.hpp>
+#include <boost/geometry/util/math.hpp>
+#include <boost/geometry/util/normalize_spheroidal_coordinates.hpp>
+
+#include <boost/geometry/algorithms/assign.hpp>
+#include <boost/geometry/algorithms/convert.hpp>
+
+#include <boost/geometry/algorithms/detail/envelope/range.hpp>
+#include <boost/geometry/algorithms/dispatch/envelope.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace envelope
+{
+
+
+class envelope_multipoint_on_spheroid
+{
+private:
+    template <typename T>
+    struct lon_lat_pair
+        : std::pair<T, T>
+    {
+        lon_lat_pair(T const& first, T const& second)
+            : std::pair<T, T>(first, second)
+        {}
+    };
+
+    struct longitude_less
+    {
+        template <typename Pair>
+        inline bool operator()(Pair const& pair1, Pair const& pair2) const
+        {
+            return math::smaller(pair1.first, pair2.first);
+        }
+    };
+
+    struct latitude_less
+    {
+        template <typename Pair>
+        inline bool operator()(Pair const& pair1, Pair const& pair2) const
+        {
+            return math::smaller(pair1.second, pair2.second);
+        }
+    };
+
+
+    template <typename Constants, typename MultiPoint, typename OutputIterator>
+    static inline void analyze_point_coordinates(MultiPoint const& multipoint,
+                                                 bool& has_south_pole,
+                                                 bool& has_north_pole,
+                                                 OutputIterator oit)
+    {
+        typedef typename boost::range_iterator
+            <
+                MultiPoint const
+            >::type iterator_type;
+        typedef typename coordinate_type<MultiPoint>::type coordinate_type;
+
+        // analyze point coordinates:
+        // (1) normalize point coordinates
+        // (2) check if any point is the north or the south pole
+        // (3) put all non-pole points in a container
+        //
+        // notice that at this point in the algorithm, we have at
+        // least two points on the spheroid
+        has_south_pole = has_north_pole = false;
```

Comment:
We normally don't do onlined multiple assignments in the library, I prefer to split